### ISSUE TITLE
boost::utility::string_view should name a type

### ISSUE THREAD

**Issue by vinniefalco** [boostorg/utility#78]
if you include `<boost/utility/string_view.hpp>` then this line should work:

```
using string_view = boost::utility::string_view;
```

**Comment by Lastique** [comment]
Why?

**Comment by glenfe** [comment]
We don't use the `utility` namespace

**Comment by vinniefalco** [comment]
> Why?

Because I can't be bothered to remember whether or not each of N Boost libraries "uses its namespace" or not. I include a header, I expect to find the symbol in its corresponding namespace. I thought the convention was that Boost.X has a namespace called X and its symbols go there. If the types are worthy they are also lifted into the boost namespace. Having random exceptions makes things harder to use.

**Comment by glenfe** [comment]
It's true for all new Boost libraries, but even (for example), `shared_ptr` is `boost::shared_ptr` not `boost::smart_ptr::shared_ptr`.

**Comment by vinniefalco** [comment]
So all new boost libraries put their symbols in `boost::`? That's news to me. I'll change Beast in time for 1.78 then.

**Comment by glenfe** [comment]
No, requiring a library namespace under `boost` is true for all new libraries. But Utility, Smart_Ptr etc. and older libraries are not subject to that rule.

**Comment by vinniefalco** [comment]
Well, that's my point - I have to remember which libraries follow the rule and which don't. I'd rather not.

**Comment by Lastique** [comment]
Libraries that focus on a given topic do use a dedicated namespace, and are supposed to in the future. Boost.Utility, Boost.Detail and to a degree Boost.Core are not such libraries - those are just a collection of random unrelated tools. Historically, those were introduced in namespace `boost`. We might consider moving some components to dedicated namespaces where it makes sense (e.g. Boost.SmartPtr). But having `boost::utility` in particular, doesn't make much sense, IMHO.

That said, it may make sense to extract `string_view` and `string_ref` to a separate library and use a namespace in *that* library.

**Comment by alexsandraaddisen26** [comment]
It would be beneficial to maintain consistency in naming conventions across Boost libraries, especially for commonly used types like `string_view`. This would help avoid confusion when using headers like `<boost/utility/string_view.hpp>` and improve code readability by aligning with user expectations regarding namespace usage.
